src/app/core/custom-encoder/custom-encoder.component.ts

```diff
@@ -0,0 +1,16 @@
+import { HttpParameterCodec } from '@angular/common/http'
+
+export class CustomEncoderComponent implements HttpParameterCodec {
+  encodeKey(key: string): string {
```

Comment:
This is a great solution! Do you think it makes sense to change the postfix from `.component.ts ` to `.encoder.ts`? since this is not a component or is not a service, but just a class

Reply:
Also, remove the testing... I believe it would break because it won't be able to `compileComponents` or `createComponent`  a component.

Reply:
@leomendoza123 Yes, it's sound better. encoder.ts. Ok, I will remove the test.